portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -34,23 +34,44 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);;
-
+    int commentCount = getRequestedCommentCount(request);
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
+    int readCommentCount = 0;
     ArrayList<String> commentSection = new ArrayList<String>();
     for (Entity entity : results.asIterable()) {
+        if (readCommentCount >= commentCount) {
+            break;
+        }
+        readCommentCount++;
         String text = (String) entity.getProperty("text");
-
-        String comment = text;
-        commentSection.add(comment);
+        commentSection.add(text);
     }
     
+    //ArrayList<String> limitedComments = new ArrayList<String>();
+    //for (int i = 0; i < commentCount; i++) {
+      //  limitedComments.add(commentSection.get(i));
+    //}
```

Comment:
Make sure you delete this code if you're not using it.

Reply:
yes, sorry. I just deleted it on a different branch, I'll delete it here too.